### ISSUE TITLE
Duplicate entries for the same module generated in app.manifest when building self-contained

### ISSUE THREAD

**Issue by dblohm7** [microsoft/WindowsAppSDK#4713]
### Describe the bug

I'm building a self-contained module that won't load due to Fusion errors. Its assembly manifest contains multiple `asmv3:file` elements for `Microsoft.UI.Xaml.dll`, however the second, duplicate entry is listed as `Microsoft.ui.xaml.dll` (notice the difference in case). AFAICT the cause is that the implementation of `GenerateAppManifestFromAppx` is not dealing with case insensitivity of Windows filesystems when it removes the name of a processed DLL file from `dllFiles`.

The error message in the event log is `Activation context generation failed for "foo.dll". Dll redirector contributor unable to add file map entry for file Microsoft.ui.xaml.dll; Two or more components referenced directly or indirectly by the application manifest have files by the same name.`

### Steps to reproduce the bug

1. Publish to filesystem a NativeAOT, unpackaged, self-contained DLL that uses the Windows SDK. Runtime identifier set to `win-x64`.
2. Try to load that DLL. The DLL load fails with the `SideBySide` entry in the event log as described above.

### Expected behavior

There should not be duplicate entries in the app manifest for the same file.

### NuGet package version

Windows App SDK 1.6.0: 1.6.240829007

### Packaging type

Unpackaged

### Windows version

Windows 11 version 22H2 (22621, 2022 Update)

### IDE

Visual Studio 2022

**Comment by Kurpanik** [comment]
Same problem here

![Image](https://github.com/user-attachments/assets/581b5399-d236-4462-8457-8d6ffc7a3787)

**Comment by c4rlosmarin** [comment]
I'm able to reroduce this issue. Got the very same error message while trying to execute a self-contained application through its .exe file.

**Comment by Scottj1s** [comment]
@dblohm7 Thanks for reporting this.  It will be fixed in the next experimental WinAppSDK 1.7 release.  To unblock, you can copy the updated task definition into your project file:

```
   <UsingTask TaskName="GenerateAppManifestFromAppx" TaskFactory="RoslynCodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll">
        <ParameterGroup>
            <RedirectDlls ParameterType="System.Boolean" Required="true" />
            <MsixContentDir ParameterType="System.String" Required="true" />
            <InAppxManifest ParameterType="System.String" Required="true" />
            <OutAppManifest ParameterType="System.String" Required="true" />
        </ParameterGroup>
        <Task>
            <Using Namespace="System.Collections.Generic" />
            <Using Namespace="System.IO" />
            <Using Namespace="System.Linq" />
            <Using Namespace="System.Text" />
            <Using Namespace="System.Xml" />
            <Code Type="Fragment" Language="cs">
                <![CDATA[
            var headerF = @"<?xml version='1.0' encoding='utf-8' standalone='yes'?>
<assembly manifestVersion='1.0'
    xmlns:asmv3='urn:schemas-microsoft-com:asm.v3'
    xmlns:winrtv1='urn:schemas-microsoft-com:winrt.v1'
    xmlns='urn:schemas-microsoft-com:asm.v1'>";
            var sb = new StringBuilder();
            sb.AppendLine(headerF);
            
            var dllFileFormat = RedirectDlls ? 
                @"    <asmv3:file name='{0}' loadFrom='%MICROSOFT_WINDOWSAPPRUNTIME_BASE_DIRECTORY%{0}'>" :
                @"    <asmv3:file name='{0}'>";
            if (!string.IsNullOrEmpty(InAppxManifest))
            {
                XmlDocument doc = new XmlDocument();
                doc.Load(InAppxManifest);
                var nsmgr = new XmlNamespaceManager(doc.NameTable);
                nsmgr.AddNamespace("m", "http://schemas.microsoft.com/appx/manifest/foundation/windows10");
                var xQuery = "./m:Package/m:Extensions/m:Extension/m:InProcessServer";
                var dllFiles = (from di in (new DirectoryInfo(MsixContentDir).EnumerateFiles("*.dll")) select di.Name).ToList();
                foreach (XmlNode winRTFactory in doc.SelectNodes(xQuery, nsmgr))
                {
                    var dllFileNode = winRTFactory.SelectSingleNode("./m:Path", nsmgr);
                    var dllFile = dllFileNode.InnerText;
                    var typesNames = winRTFactory.SelectNodes("./m:ActivatableClass", nsmgr).OfType<XmlNode>();
                    sb.AppendFormat(dllFileFormat, dllFile);
                    sb.AppendLine();
                    foreach (var typeNode in typesNames)
                    {
                        var attribs = typeNode.Attributes.OfType<XmlAttribute>().ToArray();
                        var typeName = attribs
                            .OfType<XmlAttribute>()
                            .SingleOrDefault(x => x.Name == "ActivatableClassId")
                            .InnerText;
                        var xmlEntryFormat =
@"        <winrtv1:activatableClass name='{0}' threadingModel='both'/>";
                        sb.AppendFormat(xmlEntryFormat, typeName);
                        sb.AppendLine();
                        dllFiles.RemoveAll(e => e.Equals(dllFile, StringComparison.OrdinalIgnoreCase));
                    }
                    sb.AppendLine(@"    </asmv3:file>");
                }
                if(RedirectDlls)
                {
                    foreach (var dllFile in dllFiles)
                    {
                        sb.AppendFormat(dllFileFormat, dllFile);
                        sb.AppendLine(@"</asmv3:file>");
                    }
                }
            }
            sb.AppendLine(@"</assembly>");
            var manifestContent = sb.ToString();
            File.WriteAllText(OutAppManifest, manifestContent, Encoding.UTF8);
]]>
            </Code>
        </Task>
    </UsingTask>

```
